.github/workflows/build-main.yml

```diff
@@ -55,48 +59,43 @@ jobs:
       with:
         run: python -m pytest --cov-report xml --cov=napari_aicsimageio napari_aicsimageio/tests/
     - name: Upload codecov
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v3
 
   lint:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: 3.9
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install .[test]
-    - name: Lint with flake8
-      run: |
-        flake8 napari_aicsimageio --count --verbose --show-source --statistics
-        isort napari_aicsimageio --check-only
-        mypy napari_aicsimageio
-        black --check --exclude vendor napari_aicsimageio
+        pip install pre-commit
+    - name: Lint with pre-commit
+      run: pre-commit run --all-files
 
   publish:
     if: "contains(github.event.head_commit.message, 'Bump version')"
-    needs: [test, lint]
+    needs: [check-manifest, test, lint]
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: 3.9
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install setuptools wheel
+        pip install build wheel
     - name: Build Package
-      run: |
-        python setup.py sdist bdist_wheel
```

Comment:
invoking `python setup.py` directly is deprecated.  `python -m build` is the way to go: https://github.com/pypa/build

`python -m build` will also work if you have a `setup.py` file (so the move to pyproject.toml here is slightly tangential to also using `python -m build` to actually general your sdist and wheel)

Excellent article here: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html